wayback-webapp/src/main/webapp/WEB-INF/query/BubbleCalendarHeatmap.jsp

```diff
@@ -0,0 +1,581 @@
+
+            callbacks:{
+                onUpdate:function(){
+                    $("#mCSB_1_container").css("left", x);
+                },
+                whileScrolling:function()
+                {
+                    leftAmount = $("#mCSB_1_container").css("left");
+                    sessionStorage.setItem("scrollbarX", leftAmount);
+                }
+            }
+        });
+    })(jQuery);
+</script>  
+
+<div class="clearfix"></div>
+
+<div id="wbCalendar">
+    
+  <div id="calUnder" class="calPosition">
+
+<%!
+int min = 0;
+int max = 0;
+int span = 0;
+%>
+<%
+    for(int moy = 0; moy < 12; moy++) 
+    {
+        Partition<Partition<CaptureSearchResult>> curMonth = data.monthsByDay.get(moy);
+	List<Partition<CaptureSearchResult>> monthDays = curMonth.list();
+        cal.setTime(curMonth.getStart());
+        int daysInMonth = cal.getActualMaximum(Calendar.DAY_OF_MONTH);
+
+        for(int dom = 0; dom < daysInMonth; dom++) 
+        {
+            int counter = monthDays.get(dom).count();
+
+            if(counter > 0)
+            {
+                if(counter > max)
+                {
+                    max = counter;
+
+                    if(min == 0)
+                        min = counter;
+                }
+                else if(counter < min)
+                    min = counter;
+            }
+        }
+    }
+
+span = max - min + 1;
+
+// draw 12 months, 0-11 (0=Jan, 11=Dec)
+for(int moy = 0; moy < 12; moy++) {
+	Partition<Partition<CaptureSearchResult>> curMonth = data.monthsByDay.get(moy);
+	List<Partition<CaptureSearchResult>> monthDays = curMonth.list();
+%>
+    <div class="month" id="<%= data.yearNum %>-<%= moy %>">
+	    <table>
+	
+	       <thead>
+	           <tr>
+	               <th colspan="7"><span class="label"></span></th>
+	           </tr>
+	       </thead>
+	       <tbody>
+	           <tr>
+<%
+		cal.setTime(curMonth.getStart());
+		int skipDays = cal.get(Calendar.DAY_OF_WEEK) - 1;
+		int daysInMonth = cal.getActualMaximum(Calendar.DAY_OF_MONTH);
+		// skip until the 1st:
+		for(int i = 0; i < skipDays; i++) {
+			%><td><div class="date"></div></td><%
+		}
+		int dow = skipDays;
+		int dom;
+		for(dom = 0; dom < daysInMonth; dom++) {
+
+
+			int count = monthDays.get(dom).count();
+			if(count > 0) {
+				// one or more captures in this day:
+				CaptureSearchResult firstCaptureInDay = 
+					monthDays.get(dom).list().get(0);
+				String replayUrl = uriConverter.makeReplayURI(
+							firstCaptureInDay.getCaptureTimestamp(),
+							firstCaptureInDay.getOriginalUrl());
+				Date firstCaptureInDayDate = firstCaptureInDay.getCaptureDate();
+				String safeUrl = fmt.escapeHtml(replayUrl);
+                                double alpha = (double) (count - min + 1) / span;
+
+                                if(span >= 5)
+                                {
+                                    int step = span / 5;
+
+                                    if( count < (min + step) )
```

Comment:
To be consistent with style in this file, there should not be a space following an opening parentheses or preceding a closing parentheses. Same comment for line 386, 397, and 548.